test/presubmit-tests.sh

```diff
@@ -15,6 +15,9 @@
 # limitations under the License.
 
 # This script runs the presubmit tests; it is started by prow for each PR.
+# When running `--integration-tests`, use `K8S_CLUSTER_OVERRIDE` to point at your
```

Comment:
As explained, this is not true, remove.

Reply:
in the above conversation i think we established that you _do_ need `K8S_CLUSTER_OVERRIDE` to use a predefined cluster:

![image](https://user-images.githubusercontent.com/432502/46436869-d6157480-c70e-11e8-8673-ae0800499997.png)

Reply:
There are 3 different contexts here:
* running on Prow (line 17), you don't need to provide anything
* running locally, you can just set PROJECT_ID and the test will create a new cluster for you
* running locally, you can set K8S_* **and** use `--run-tests` **but not with `presubmit-tests.sh`** to run the tests against your cluster

This is only true because you hardcoded `--run-tests` in your script, which is not needed.

Reply:
> running locally, you can set K8S_* and use --run-tests but not with presubmit-tests.sh to run the tests against your cluster

hm so I'd have to provide another entrypoint that doesn't source `presubmit-tests.sh` for running locally?

i'm kind of okay with the hack for now, if i can keep the same entrypoint

Reply:
Christie, the root cause of all this confusion is that you're implementing the scripts in a way that the helpers were not intended to be used. This is causing all sort of hacks, confusions and wasted work (like debugging and unnecessary changes in knative/test-infra).

If you just follow the recipe used by all the other repos, things will be easier, you won't need any hacks and you'll have all that you want: presubmit, integration tests on prow and integration tests locally.

Reply:
In https://github.com/knative/build-pipeline/pull/87/commits/60e4470183dc4261ffd7ee328ea32166f7ea4f13 I've made my best attempt to use the helpers in the way they were intended to be used, and it seems to have helped, but there was really only one thing I was doing differently, which was that I didn't have a script called `e2e-tests.sh` which invoked my tests.

I was doing my best to comply with what the other repos were doing, but did not expect that the name of the files I was creating was part of the interface, or that some unseen mechanism (which I still can't track down - where is the boskos logic?) would look `e2e-tests.sh` and call it with `--run-tests`.

The docs you added at https://github.com/knative/test-infra/blob/master/scripts/README.md will go a long way to helping clear this up, but even seeing those docs (when they were in the script itself), I still didn't understand how this was going to be called, and looking at the Prow configs wasn't enough.

It would go a long way to making this easier to be able to easily trace through exactly what code will be executed by Prow (and by the image Prow is calling?).

> This is causing all sort of hacks, confusions and wasted work (like debugging and unnecessary changes in knative/test-infra).

As far as I can tell, the cluster creation scripts still won't work without `BUILD_NUMBER` being defined, and that caused the issue that I mis-diagnosed and tried to fix in test-infra.